### PR TITLE
[feat:#84][client]:pb define, broker tcp server

### PR DESCRIPTION
**Pull request by XiaTianliang**

### CODE REVIEW

File: storage/handler/writer.go [hunk 1]
```diff
@@ -135,26 +124,57 @@ func (w *Writer) Write(stream storage.WriteService_WriteServer) error {
 	}
 }
 
+func (w *Writer) handleReplica(replica *storage.Replica) {
+	reader := streamIO.NewReader(replica.Data)
+	for !reader.Empty() {
```
Comment (CodingCrush) [inline storage/handler/writer.go]: there are 2 ways to reduce the test miss here:
1. you can use `for !reader.Empty() && reader.Err() == nil`
2. check reader.Err() after reading all data, it is not necessary to check error each step, it's too hard to mock every errors for data corruption

File: tsdb/memdb/metric_store.go [hunk 2]
```diff
@@ -152,6 +152,29 @@ func (ms *metricStore) getMetricID() uint32 {
 	return atomic.LoadUint32(&ms.metricID)
 }
 
+// convert tags to string
```
Comment (CodingCrush) [inline tsdb/memdb/metric_store.go]: move this method to another package, such as `model`?

File: tsdb/memdb/metric_store.go [hunk 3]
```diff
@@ -152,6 +152,29 @@ func (ms *metricStore) getMetricID() uint32 {
...
+	}
+
+	sort.Strings(tagKeyValues)
+
+	var builder strings.Builder
+	builder.Grow(totalLen)
+
+	for _, tag := range tagKeyValues {
+		builder.WriteString(tag)
+	}
```
Comment (CodingCrush) [inline tsdb/memdb/metric_store.go]: writing tagValue is missing

File: rpc/server.go [hunk 4]
```diff
@@ -3,55 +3,124 @@ package rpc
...
+	}
+
+	s.lis = lis
+	s.logger.Info("rpc tcpServer start serving", logger.String("address", s.bindAddress))
+
+	for {
+		// Listen for an incoming connection.
+		conn, err := lis.Accept()
+		if err != nil {
+			s.logger.Error("tcp server error when accepting", zap.Error(err))
```
[the remark this answers is not in the document]
Comment (XiaTianliang) [reply rpc/server.go]: ok

File: rpc/server.go [hunk 5]
```diff
@@ -3,55 +3,124 @@ package rpc
...
+type grpcServer struct {
+	bindAddress string
+	logger      *logger.Logger
+	gs          *grpc.Server
+}
+
+func NewGRPCServer(bindAddress string) GRPCServer {
+	return &grpcServer{
 		bindAddress: bindAddress,
+		logger:      logger.GetLogger("rpc", "grpcServer"),
```
[the remark this answers is not in the document]
Comment (XiaTianliang) [reply rpc/server.go]: ok

**Pull request merged**